lexers/perl.lua

```diff
@@ -102,25 +66,31 @@ local lit_str = 'q' * P('q')^-1 * literal_delimited
 local lit_array = 'qw' * literal_delimited
 local lit_cmd = 'qx' * literal_delimited
 local lit_tr = (P('tr') + 'y') * literal_delimited2 * S('cds')^0
-local string = token(lexer.STRING,
+local string = lex:tag(lexer.STRING,
   sq_str + dq_str + cmd_str + heredoc + lit_str + lit_array + lit_cmd + lit_tr)
 local regex_str = lexer.after_set('-<>+*!~\\=%&|^?:;([{', lexer.range('/', true) * S('imosx')^0)
 local lit_regex = 'qr' * literal_delimited * S('imosx')^0
 local lit_match = 'm' * literal_delimited * S('cgimosx')^0
 local lit_sub = 's' * literal_delimited2 * S('ecgimosx')^0
-local regex = token(lexer.REGEX, regex_str + lit_regex + lit_match + lit_sub)
+local regex = lex:tag(lexer.REGEX, regex_str + lit_regex + lit_match + lit_sub)
 lex:add_rule('string', string + regex)
 
+-- Functions.
```

Comment:
What was the reason for switching the rule order here? Is there something about strings that prevents them from matching after functions/methods? (I do not know much about Perl.)

Reply:
During testing the old order broke regex strings. The easiest fix was adjusting the rule order.
Perl being Perl it still might break in unexpected ways.